src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/Program.cs

```diff
@@ -84,6 +84,8 @@ static async Task RunAsync(ServerConfiguration serverConfiguration, Cancellation
     var typeRefResolver = new ExtensionTypeRefResolver(assemblyLoader, loggerFactory);
 
     using var exportProvider = await ExportProviderBuilder.CreateExportProviderAsync(extensionManager, assemblyLoader, serverConfiguration.DevKitDependencyPath, loggerFactory);
+    // Ensure the ExtensionAssemblyManager is available before we initialize anything else.
+    exportProvider.GetExportedValue<ExtensionAssemblyManagerMefProvider>().SetMefExtensionAssemblyManager(extensionManager);
```

Comment:
Consider moving this to:

https://github.com/dotnet/roslyn/blob/b9236353c566fa7c8d13a4b147ab964afb1c4323/src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/ExportProviderBuilder.cs#L69-L70

Since we're already doing the same pattern there too. That way it's consistent that stuff that went into CreateExportProviderAsync gets set on the return.